DFE-Digital/sts-plan-technology-for-your-school
# fix: Create empty workflow we can use to verify checks

## Overview

Addresses ticket [#220750](https://dfe-ssp.visualstudio.com/s190-schools-technology-services/_boards/board/t/Plan%20Tech/Stories/?workitem=220750)

Addresses the issue of failing code being mergeable when the autoformatter has run and removed checks

## Changes

creates an empty stub workflow that will run on any push/pr on main and dev that we can set as a mandatory status check in branch rules

## How to review the PR

Test steps:
- we will probably have to merge this to actually test it properly
- Create a new branch off development and make a change that will both 
  - break the tests
  - make the autoformatter run 
  - (e.g. edit an assertion to make it fail, and put it on the same line as the previous one)
- Wait for the autoformatter

The status checks will disappear because the autoformat cannot trigger another workflow (so the on push won't run again) 
but if this stub workflow with branch rules has worked, we should be unable to merge, because those rules will note that the stubbed workflow hasn't run.

We should also test that the new branch ruleset doesn't override any important configuration in our existing classic protection rules

## Additional notes

This will require setting up a branch ruleset which requires status checks and 
"Verify Checks Running" 
needs adding to the list 

## Checklist

- [ ] Branch rulesets have been updated
